bemb/model/bemb.py

```diff
@@ -1313,18 +1341,21 @@ def elbo(self, batch: ChoiceDataset, num_seeds: int = 1) -> torch.Tensor:
         # ==============================================================================================================
         if self.pred_item:
             # the prediction target is item_index.
-            elbo += self.forward(batch,
+            elbo_expanded = self.forward(batch,
```

Comment:
why don't we just call the `forward()` function with `deterministic=self.deterministic_variational`? With `forward(deterministic=True)`, the `forward()` function will disregard the `sample_dict` passed in, it will creates another copy of `sample_dict` with `num_seeds=1` and fill this dict with *mean* of parameters.

```
        if deterministic:
            num_seeds = 1
            # Use the means of variational distributions as the sole deterministic MC sample.
            # NOTE: here we don't need to sample the obs2prior weight H since we only compute the log-likelihood.
            # TODO: is this correct?
            sample_dict = dict()
            for coef_name, coef in self.coef_dict.items():
                sample_dict[coef_name] = coef.variational_distribution.mean.unsqueeze(
                    dim=0)  # (1, num_*, dim)
```

Reply:
we can do that as well, but we do need the sample dict earlier to send to the `log_prior()` (same response to next question)